src/include/daos_errno.h

```diff
@@ -43,143 +43,209 @@ extern "C" {
+	ACTION(DER_NOTATTACH,		(DER_ERR_GURT_BASE + 30),	\
+	       Service group not attached)				\
 	/** version mismatch */						\
-	ACTION(DER_MISMATCH,		(DER_ERR_GURT_BASE + 31))	\
+	ACTION(DER_MISMATCH,		(DER_ERR_GURT_BASE + 31),	\
+	       Version mismatch)					\
 	/** rank has been evicted */					\
-	ACTION(DER_EVICTED,		(DER_ERR_GURT_BASE + 32))	\
+	ACTION(DER_EVICTED,		(DER_ERR_GURT_BASE + 32),	\
+	       Rank has been evicted)					\
 	/** user-provided RPC handler didn't send reply back */		\
-	ACTION(DER_NOREPLY,		(DER_ERR_GURT_BASE + 33))	\
+	ACTION(DER_NOREPLY,		(DER_ERR_GURT_BASE + 33),	\
+	       User provided RPC handler did not send reply back)	\
 	/** denial-of-service */					\
-	ACTION(DER_DOS,			(DER_ERR_GURT_BASE + 34))       \
+	ACTION(DER_DOS,			(DER_ERR_GURT_BASE + 34),       \
+	       Denial of service)					\
 	/** Incorrect target for the RPC  */				\
-	ACTION(DER_BAD_TARGET,		(DER_ERR_GURT_BASE + 35))	\
+	ACTION(DER_BAD_TARGET,		(DER_ERR_GURT_BASE + 35),	\
+	       Incorrect target for the RPC)				\
 	/** Group versioning mismatch */				\
-	ACTION(DER_GRPVER,		(DER_ERR_GURT_BASE + 36))
+	ACTION(DER_GRPVER,		(DER_ERR_GURT_BASE + 36),	\
+	       Group versioning mismatch)
 	/** TODO: add more error numbers */
 
 #define D_FOREACH_DAOS_ERR(ACTION)					\
 	/** Generic I/O error */					\
-	ACTION(DER_IO,			(DER_ERR_DAOS_BASE + 1))	\
+	ACTION(DER_IO,			(DER_ERR_DAOS_BASE + 1),	\
+	       Input / output error)					\
 	/** Memory free error */					\
-	ACTION(DER_FREE_MEM,		(DER_ERR_DAOS_BASE + 2))	\
+	ACTION(DER_FREE_MEM,		(DER_ERR_DAOS_BASE + 2),	\
+	       Memory free error)					\
 	/** Entry not found */						\
-	ACTION(DER_ENOENT,		(DER_ERR_DAOS_BASE + 3))	\
+	ACTION(DER_ENOENT,		(DER_ERR_DAOS_BASE + 3),	\
+	       Entity not found)					\
 	/** Unknown object type */					\
-	ACTION(DER_NOTYPE,		(DER_ERR_DAOS_BASE + 4))	\
+	ACTION(DER_NOTYPE,		(DER_ERR_DAOS_BASE + 4),	\
+	       Unknown object type)					\
 	/** Unknown object schema */					\
-	ACTION(DER_NOSCHEMA,		(DER_ERR_DAOS_BASE + 5))	\
+	ACTION(DER_NOSCHEMA,		(DER_ERR_DAOS_BASE + 5),	\
+	       Unknown object schema)					\
 	/** Object is not local */					\
-	ACTION(DER_NOLOCAL,		(DER_ERR_DAOS_BASE + 6))	\
+	ACTION(DER_NOLOCAL,		(DER_ERR_DAOS_BASE + 6),	\
+	       Object is not local)					\
 	/** stale pool map version */					\
-	ACTION(DER_STALE,		(DER_ERR_DAOS_BASE + 7))	\
+	ACTION(DER_STALE,		(DER_ERR_DAOS_BASE + 7),	\
+	       Stale pool map version)					\
 	/** Not service leader */					\
-	ACTION(DER_NOTLEADER,		(DER_ERR_DAOS_BASE + 8))	\
+	ACTION(DER_NOTLEADER,		(DER_ERR_DAOS_BASE + 8),	\
+	       Not service leader)					\
 	/** * Target create error */					\
-	ACTION(DER_TGT_CREATE,		(DER_ERR_DAOS_BASE + 9))	\
+	ACTION(DER_TGT_CREATE,		(DER_ERR_DAOS_BASE + 9),	\
+	       Target create error)					\
 	/** Epoch is read-only */					\
-	ACTION(DER_EP_RO,		(DER_ERR_DAOS_BASE + 10))	\
+	ACTION(DER_EP_RO,		(DER_ERR_DAOS_BASE + 10),	\
+	       Epoch is read - only)					\
```

Comment:
remove -.   Is there any benefit in not simply using strings for these parameters?

Reply:
Not sure about that. Perhaps a compact notation. My idea was to reuse the already existing code by expanding the macro. On the not-so-bright side, macros might be a little bit complicated.